### PR TITLE
Fix issue where colorbar moves second x axis

### PR DESCRIPTION
**Pull request by Jaroza727**
## PR Summary
Resolves issue [#5904](https://github.com/matplotlib/matplotlib/issues/5904). This issue has an existing patch; this pull request implements the patch and adds a test case.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [N/A ] New features are documented, with examples if plot related.
- [N/A ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [N/A ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/matplotlib/tests/test_axes.py [hunk 1]
```diff
@@ -34,6 +34,7 @@
     assert_allclose, assert_array_equal, assert_array_almost_equal)
 from matplotlib import rc_context
 from matplotlib.cbook import MatplotlibDeprecationWarning
+from mpl_toolkits.axes_grid1 import make_axes_locatable
```
Comment (jklymak) [inline lib/matplotlib/tests/test_axes.py:37]: this cannot go in the main library.  Please test in `mpl_toolkits/tests/`  Thanks!

**Review by jklymak — DISMISSED** [review]

File: lib/mpl_toolkits/tests/test_axes_grid1.py [hunk 2]
```diff
@@ -478,3 +478,32 @@ def test_axes_class_tuple():
...
+
+@image_comparison(['second_axis_with_colorbar.png'])
+def test_second_axis_with_colorbar():
+    X = np.array([1, 2, 3, 4])
+    X2 = np.array([1, 2, 3, 4])
+    Y = np.cos(X*20)
+    Z = np.sin(X*20)
+
+    fig = plt.figure()
+    ax1 = plt.subplot(111)
```
Comment (QuLogic) [inline lib/mpl_toolkits/tests/test_axes_grid1.py:491]:
```suggestion
    fig, ax1 = plt.subplots()
```

File: lib/matplotlib/tests/test_axes.py [hunk 3]
```diff
@@ -6824,3 +6824,4 @@ def test_ylabel_ha_with_position(ha):
     ax.set_ylabel("test", y=1, ha=ha)
     ax.yaxis.set_label_position("right")
     assert ax.yaxis.get_label().get_ha() == ha
+
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_axes.py:6827]: This must be reverted as well.

**Comment by QuLogic** [comment]
Closing as per #5904 and #22667.

**Pull request closed without merging**